### ISSUE TITLE
2.5.5 npm release contains .git folder

### ISSUE THREAD

**Issue by samueljmurray** [react-native-community/react-native-linear-gradient#429]
# Bug

Since earlier today, when installing npm packages, we get the following error:

```bash
NPM err! path node_modules/react-native-linear-gradient: Appears to be a git repo or submodule.
```

## Environment info

React native info output:

```bash
info
  React Native Environment Info:
    System:
      OS: macOS 10.14.4
      CPU: (4) x64 Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz
      Memory: 108.42 MB / 16.00 GB
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 9.7.1 - /usr/local/bin/node
      Yarn: 1.5.1 - /usr/local/bin/yarn
      npm: 5.6.0 - /usr/local/bin/npm
      Watchman: 4.5.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.2, macOS 10.14, tvOS 12.2, watchOS 5.2
      Android SDK:
        API Levels: 23, 24, 25, 26, 27, 28
        Build Tools: 23.0.1, 23.0.2, 23.0.3, 25.0.0, 25.0.1, 25.0.2, 25.0.3, 26.0.1, 26.0.3, 27.0.3, 28.0.0, 28.0.3
        System Images: android-23 | Intel x86 Atom, android-23 | Intel x86 Atom_64, android-27 | Google APIs Intel x86 Atom, android-27 | Google Play Intel x86 Atom, android-28 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5264788
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.3 => 16.8.3
      react-native: ^0.59.10 => 0.59.10
    npmGlobalPackages:
      create-react-native-app: 1.0.0
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7
```

Library version: 2.5.5

## To reproduce

1. Run `npm install --save react-native-linear-gradient`

2. Run `npm install`

**Comment by AngefloMusic** [comment]
+1 on this new installs without a lock to 2.5.4 are failing for me

**Comment by daxaxelrod** [comment]
Same here

**Comment by jedashford** [comment]
Our process is broken because of this, not able to install new packages.

**Comment by AngefloMusic** [comment]
@jedashford run ```npm install --save react-native-linear-gradient@2.5.4 ``` as a temporary measure

**Comment by byk04712** [comment]
same error here

**Comment by HarisSpahija** [comment]
break happened with this merge #417

**Comment by mikehardy** [comment]
I believe the problem is just an incorrect packaging of the repo on the way to upload to npmjs.org, I don't believe it has anything to do with #417 

A workaround is possible with a postinstall in your package.json `rm -fr ./node_modules/react-native-linear-gradient/.git`:

```json
  ...
  "postinstall": "rm -fr ./node_modules/react-native-linear-gradient/.git",
  ...
```

I wouldn't presume to tell other maintainers how to release their code (there are infinite good ways to do it) but I have good luck with the ['np' package](https://www.npmjs.com/package/np) like so https://github.com/react-native-community/react-native-device-info/blob/master/package.json#L16 - so that I always package and release the same way. Another valid way that makes me jealous of other libraries when I see them integrate it well is to use semantic-release / semantic-release-bot etc.

**Comment by callmejm** [comment]
for temporary solve , go to your `package.json` in `script` add a new command like `"rmgit": "rm -r node_modules/react-native-linear-gradient/.git"` , then in your root console `npm run rmgit`

**Comment by vovkasm** [comment]
I suppose it is this story: https://npm.community/t/npm-6-9-1-is-broken-due-to-git-folder-in-published-tarball/8454/15
More concrete bug: https://github.com/npm/npm-packlist/issues/35

**Comment by vovkasm** [comment]
So probably fix would be simple publish new version with more new npm client (more fresh npm-packlist package). Publisher can test what will be in archive by `npm pack --dry-run`

**Comment by mikehardy** [comment]
@vovkasm good find! wow what a pain. I just published a couple revs of other libraries and I suppose I just got lucky I wasn't fully up to date w.r.t. npm

**Comment by vovkasm** [comment]
Yes, just wonder, can we tag v2.5.6 and publish it to npm?

**Comment by dimlksin2** [comment]
Same here

**Comment by HarisSpahija** [comment]
So the fix is basically internal NPM update for the maintainers? Is there any way we/maintainers could publish a hotfix?

**Comment by mikehardy** [comment]
@dimlksin2 +1 and "same here" are just comment spam in the open source world. Please just add a reaction to the issue and/or subscribe if you have nothing to move the issue forward

And yes, this will remain broken until the maintainers here use a non-broken version of npm to pack a release and upload it to npmjs.org

**Comment by radex** [comment]
I think I fixed the issue. Let me know if not fixed.

**Comment by ethanysc** [comment]
Still seeing the same errors on 2.5.5

**Comment by vovkasm** [comment]
@ethanysc As I can see, it was fixed in 2.5.6 release, 2.5.5 wasn't re-uploaded.
